src/udp_wrap.cc

```diff
@@ -446,6 +446,10 @@ void UDPWrap::OnRecv(uv_udp_t* handle,
   }
 
   char* base = static_cast<char*>(realloc(buf->base, nread));
+  // If realloc fails, fallback to old pointer
+  if(base == NULL && nread != 0)
+      base = buf->base;
```

Comment:
Also it is not specified if reallc will free memory if nread is zero. Man page said that this will be free()-abke pointer, but it may not be NULL. this may lead to memory leak. Why not to just free() memory if nread become zero?